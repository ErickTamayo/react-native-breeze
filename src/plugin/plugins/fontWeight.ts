import { PluginFunction, PluginPattern } from "./types";

export type PluginGroups = {
  key: string;
};

export const pattern: PluginPattern = ({ keys }) => {
  return new RegExp(`^font-(?<key>${keys("fontWeight")})$`);
};

export const plugin: PluginFunction<PluginGroups> = ({ groups, theme }) => {
  const { key } = groups;
  const value = theme(["fontWeight", key]);
  return { fontWeight: value };
};
