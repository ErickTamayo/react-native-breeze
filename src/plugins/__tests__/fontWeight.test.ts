import {
  generateInput,
  shouldEvaluateTheCorrectPatternTest,
  shouldMatchOutputSnapshot,
} from "../../helpers/tests";
import { pattern, plugin } from "../fontWeight";

describe("fontWeight", () => {
  // prettier-ignore
  const shouldMatch = generateInput("font", ["hairline", "thin", "light", "normal", "medium", "semibold", "bold", "extrabold", "black"]);

  shouldEvaluateTheCorrectPatternTest(pattern, {
    shouldMatch,
    shouldNotMatch: ["font-unknown"],
  });

  shouldMatchOutputSnapshot(pattern, plugin, shouldMatch);
});
