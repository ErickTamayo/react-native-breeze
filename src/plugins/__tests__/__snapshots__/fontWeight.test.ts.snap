// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fontWeight Should parse (font-black) correctly 1`] = `
Object {
  "fontWeight": "900",
}
`;

exports[`fontWeight Should parse (font-bold) correctly 1`] = `
Object {
  "fontWeight": "700",
}
`;

exports[`fontWeight Should parse (font-extrabold) correctly 1`] = `
Object {
  "fontWeight": "800",
}
`;

exports[`fontWeight Should parse (font-hairline) correctly 1`] = `
Object {
  "fontWeight": "100",
}
`;

exports[`fontWeight Should parse (font-light) correctly 1`] = `
Object {
  "fontWeight": "300",
}
`;

exports[`fontWeight Should parse (font-medium) correctly 1`] = `
Object {
  "fontWeight": "500",
}
`;

exports[`fontWeight Should parse (font-normal) correctly 1`] = `
Object {
  "fontWeight": "400",
}
`;

exports[`fontWeight Should parse (font-semibold) correctly 1`] = `
Object {
  "fontWeight": "600",
}
`;

exports[`fontWeight Should parse (font-thin) correctly 1`] = `
Object {
  "fontWeight": "200",
}
`;
